src/couch_replicator/src/couch_replicator_docs.erl

```diff
@@ -615,10 +615,10 @@ ssl_verify_options(false) ->
     [{verify, verify_none}].
 
 
--spec before_doc_update(#doc{}, Db::any()) -> #doc{}.
-before_doc_update(#doc{id = <<?DESIGN_DOC_PREFIX, _/binary>>} = Doc, _Db) ->
+-spec before_doc_update(#doc{}, Db::any(), any()) -> #doc{}.
```

Comment:
we shouldn't use `any()` here. By using `any` we disable dializer. Try defining `update_type()` in `couch_db` and refer to it using `couch_db:update_type()`.

```
diff --git a/src/couch/src/couch_db.erl b/src/couch/src/couch_db.erl
index 1f8bc42..43ed562 100644
--- a/src/couch/src/couch_db.erl
+++ b/src/couch/src/couch_db.erl
@@ -135,6 +135,10 @@
 -include_lib("couch/include/couch_db.hrl").
 -include("couch_db_int.hrl").

+-type update_type()
+    :: replicated_changes
+        | interactive_edit.
+
```